utils.py

```diff
@@ -430,3 +433,102 @@ def train_csv_loader(base_dir: Path, csv_path: Path, nids: int) -> Tuple[List[Mu
         train_dirs.append(capture_dir)
 
     return train_captures, train_dirs
+
+
+def xid_eval(
+    model,
+    driver_dataiter,
+    all_neut_avgtex_vert,
+    config,
+    output_set,
+    outpath,
+    rank,
+    iternum,
+    indices_subjects=None,
+    training=True,
+):
+    starttime = time.time()
+
+    if indices_subjects == None:
+        indices_subjects = random.sample(range(0, len(all_neut_avgtex_vert)), config.progress.cross_id_n_subjects)
+        indices_subjects.sort()
```

Comment:
This can probably be done with `np.random.permutation` or `random.choice`; see https://stackoverflow.com/questions/66899474/how-to-get-a-random-permutation-of-integers-inside-range-without-permuting-the-w